### PR TITLE
Feature/implement api layer for communications

### PR DESCRIPTION
**Pull request by the-17th-fox**
## What

Implemented API layer for Communications microservice.
Added Controllers (with related viewmodels), Configurations (extens. methods), middlewares and some utility.
Did a bit of Core,Infrastructure and Application refactoring.

## How

Added auth based controllers for every entity.
Added extension methods for ServicesCollection with infrastructure and utilities injections.
Added middlewares for db migrating and global exceptions handler.
Also added small utility for taking header value by it's key.

## Affected Area

Communications.Api
- .Controllers
- .Configuration
- .ViewModels
- .Middlewares
- .Utilities

Communications.Application
- .Notifications
- .Replies
- .Reports

Communications.Infrastructure
- .DbContext.Configurations

Communications.Core
- .Interfaces
- .Models

### CODE REVIEW

File: src/CommunicationsMicroservice/Communications.Api/Controllers/Replies/AuthoritiesRepliesController.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        private string _organization = string.Empty;
+        private Guid _userId;
+
+        public AuthoritiesRepliesController(IMediator mediator, IMapper mapper)
+        {
+            _mediator = mediator ?? throw new ArgumentNullException(nameof(mediator));
+            _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+
+        private void InitializeRequestProperties()
```
Comment (KatsiarynaAnashkevich) [inline src/CommunicationsMicroservice/Communications.Api/Controllers/Replies/AuthoritiesRepliesController.cs:28]: Private methods should be in the end of file

File: src/CommunicationsMicroservice/Communications.Api/Controllers/Reports/CiviliansReportsController.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        private string _firstName = string.Empty;
+        private string _patronymic = string.Empty;
+
+        public CiviliansReportsController(IMediator mediator, IMapper mapper)
+        {
+            _mediator = mediator ?? throw new ArgumentNullException(nameof(mediator));
+            _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+
+        private void InitializeRequestProperties()
```
Comment (KatsiarynaAnashkevich) [inline src/CommunicationsMicroservice/Communications.Api/Controllers/Reports/CiviliansReportsController.cs:31]: Private methods should be in the end of file

File: src/CommunicationsMicroservice/Communications.Api/Program.cs [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
-builder.Services.AddEndpointsApiExplorer();
-builder.Services.AddSwaggerGen();
+services.AddControllers();
+services.AddEndpointsApiExplorer();
+services.AddSwaggerGen();
 
 var app = builder.Build();
 
-// Configure the HTTP request pipeline.
 if (app.Environment.IsDevelopment())
```
Comment (KatsiarynaAnashkevich) [inline src/CommunicationsMicroservice/Communications.Api/Program.cs:17]: Could you explain this condition?

Comment (the-17th-fox) [reply src/CommunicationsMicroservice/Communications.Api/Program.cs:17]: If the app's environment variable is Development this condition returns true. This means the code within curly braces will be executed.

**Review by KatsiarynaAnashkevich — APPROVED** [review]

**Pull request merged**